NeuralAmpModeler/dsp/get_dsp.cpp

```diff
@@ -16,94 +16,106 @@ void verify_config_version(const std::string version)
     throw std::runtime_error("Unsupported config version");
 }
 
-std::unique_ptr<DSP> get_dsp(const std::filesystem::path dirname)
+std::unique_ptr<DSP> get_dsp(const std::filesystem::path dirname, dspData& returnedConfig)
 {
-  const std::filesystem::path config_filename = dirname / std::filesystem::path("config.json");
-  if (!std::filesystem::exists(config_filename))
-    throw std::runtime_error("Config JSON doesn't exist!\n");
-  std::ifstream i(config_filename);
-  nlohmann::json j;
-  i >> j;
-  verify_config_version(j["version"]);
```

Comment:
You should `verify_config_version` here. Otherwise, we don't know whether `dspData` is an appropriate struct for holding the contents that were read in.

Reply:
Done... The code does it twice now when using the get_dsp with a file argument but... Better to be safe than sorry? :D

Reply:
Mmh, yeah not ideal but should be easy to deal with. I'll take a closer look on my own time and see if there's a way to untangle.